Extensions/PathfindingBehavior/pathfindingruntimebehavior.ts

```diff
@@ -574,7 +573,9 @@ namespace gdjs {
           //Make sure we do not search forever.
           const n =
             //Get the most promising node...
-            this._openNodes.shift();
+            this._openNodes.shift() || null;
+          if (n == null)
+            return false;
```

Comment:
I'm not sure about that.

Reply:
Because of the while condition, `this._openNodes.length !== 0`, the result of `shift()` can't be undefined, as the array is not empty. So we should be able to keep the existing code there I think?

Reply:
It's because of the shift signature:
Array<gdjs.PathfindingRuntimeBehavior.Node>.shift(): Node | undefined

Maybe iterate on shift?
for (let n = this._openNodes.shift(); n; n = this._openNodes.shift())

Reply:
I would keep the while as I think it better express the intention (in A*, you continue as long as you have open nodes) but maybe just use a // @ts-ignore to tell that `invariant: _openNodes can't be empty, so n can't be null.`?

Reply:
ok, I use the "!" operator.